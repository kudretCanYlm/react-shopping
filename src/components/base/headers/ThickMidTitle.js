import PropTypes from 'prop-types';

const ThickMidTitle = ({ children, className = '' }) => {
  return <h2 className={`title-mid title-thick title-purple ${className}`}>{children}</h2>;
};

//prop-type
ThickMidTitle.propTypes = {
  children: PropTypes.object.isRequired
};

export default ThickMidTitle;
