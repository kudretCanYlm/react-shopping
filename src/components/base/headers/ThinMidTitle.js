import PropTypes from 'prop-types';

const ThinMidTitle = ({ children, className = '' }) => {
  return <h2 className={`title-mid title-thin title-purple ${className}`}>{children}</h2>;
};

//prop-type
ThinMidTitle.propTypes = {
  children: PropTypes.object.isRequired,
  className: PropTypes.string
};

export default ThinMidTitle;
