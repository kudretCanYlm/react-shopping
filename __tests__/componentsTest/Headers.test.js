import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import ThinMidTitle from 'components/base/headers/ThinMidTitle';
import ThickMidTitle from 'components/base/headers/ThickMidTitle';

Enzyme.configure({ adapter: new Adapter() });

describe('ThinMidTitleTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<ThinMidTitle className="test">Test title</ThinMidTitle>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('has my className and text', () => {
    const titleText = 'test title';
    const titleClassName = 'test className';

    const title = shallow(<ThinMidTitle className={titleClassName}>{titleText}</ThinMidTitle>);

    expect(title.hasClass(titleClassName)).toEqual(true);
    expect(title.contains(titleText)).toEqual(true);

    expect(title.hasClass('not include ClassName')).toEqual(false);
    expect(title.contains('not include text')).toEqual(false);
  });
});

describe('ThickMidTitleTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<ThickMidTitle className="test">Test component</ThickMidTitle>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('has my className and text', () => {
    const titleText = 'test title';
    const titleClassName = 'test className';

    const title = shallow(<ThickMidTitle className={titleClassName}>{titleText}</ThickMidTitle>);

    expect(title.hasClass(titleClassName)).toEqual(true);
    expect(title.contains(titleText)).toEqual(true);

    expect(title.hasClass('not include ClassName')).toEqual(false);
    expect(title.contains('not include text')).toEqual(false);
  });
});
