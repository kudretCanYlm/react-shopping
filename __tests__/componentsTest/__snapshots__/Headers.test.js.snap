// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThickMidTitleTest render correctly 1`] = `
<h2
  className="title-mid title-thick title-purple test"
>
  Test component
</h2>
`;

exports[`ThinMidTitleTest render correctly 1`] = `
<h2
  className="title-mid title-thin title-purple test"
>
  Test title
</h2>
`;
